src/lib.rs

```diff
@@ -2221,6 +2332,23 @@ enum GeneratedTrampoline {
     ResourceRep(TypeResourceTableIndex),
     /// The guest would like to drop a resource.
     ResourceDrop(TypeResourceTableIndex, Option<CoreDef>),
+    /// A Utf8 string is copied from one component's memory to the other's.
+    Utf8CopyTranscoder {
```

Comment:
This variant could be made more generic to support the other transcoding modes, e.g. by having our own op enum that only has one variant for now